tools/istio-nftables/pkg/capture/run.go

```diff
@@ -0,0 +1,1026 @@
+	}
+	if cfg.cfg.InboundInterceptionMode == "TPROXY" {
+		for _, excludeInterface := range split(cfg.cfg.ExcludeInterfaces) {
+
+			cfg.ruleBuilder.AppendRule(
+				constants.PreroutingChain, constants.IstioProxyMangleTable, "iifname", excludeInterface, "return")
+			cfg.ruleBuilder.AppendRule(constants.OutputChain, constants.IstioProxyMangleTable, "oifname", excludeInterface, "return")
+		}
+	}
+}
+
+// Run is the main function that builds and applies nftables rules based on the provided configuration.
+// It handles exclusion and inclusion logic for inbound and outbound traffic, DNS redirection,
+// owner-based filtering, TPROXY mark handling, and finally applies all the rules.
+func (cfg *NftablesConfigurator) Run() (*knftables.Transaction, error) {
+	// Since OUTBOUND_IP_RANGES_EXCLUDE could carry ipv4 and ipv6 ranges
+	// need to split them in different arrays one for ipv4 and one for ipv6
+	// in order to not to fail
+	ipv4RangesExclude, ipv6RangesExclude, err := cfg.separateV4V6(cfg.cfg.OutboundIPRangesExclude)
+	if err != nil {
+		return nil, err
+	}
+	if ipv4RangesExclude.IsWildcard {
+		return nil, fmt.Errorf("invalid value for OUTBOUND_IP_RANGES_EXCLUDE")
+	}
+
+	ipv4RangesInclude, ipv6RangesInclude, err := cfg.separateV4V6(cfg.cfg.OutboundIPRangesInclude)
+	if err != nil {
+		return nil, err
+	}
+
+	redirectDNS := cfg.cfg.RedirectDNS
+	// How many DNS flags do we have? Three DNS flags! AH AH AH AH
+	if redirectDNS && !cfg.cfg.CaptureAllDNS && len(cfg.cfg.DNSServersV4) == 0 && len(cfg.cfg.DNSServersV6) == 0 {
+		log.Warn("REDIRECT_DNS is set, but CAPTURE_ALL_DNS is false, and no DNS servers provided. DNS capture disabled.")
+		redirectDNS = false
+	}
+
+	cfg.logConfig()
+
+	// Add rules to skip specific interfaces from redirection.
+	cfg.shortCircuitExcludeInterfaces()
+	cfg.shortCircuitKubeInternalInterface()
+
+	// Create a rule for invalid drop in PREROUTING chain in mangle table, so the nftables will drop the out of window packets instead of reset connection .
+	dropInvalid := cfg.cfg.DropInvalid
+	if dropInvalid {
+		cfg.ruleBuilder.AppendRule(constants.PreroutingChain, constants.IstioProxyMangleTable,
+			"meta l4proto tcp",
+			"ct state", "invalid",
+			"jump", constants.IstioDropChain)
+		cfg.ruleBuilder.AppendRule(constants.IstioDropChain, constants.IstioProxyMangleTable, "drop")
+	}
+
+	// Create a new chain for to hit tunnel port directly. Envoy will be listening on port acting as VPN tunnel.
+	cfg.ruleBuilder.AppendRule(constants.IstioInboundChain, constants.IstioProxyNatTable,
+		"meta l4proto tcp",
+		"tcp dport", cfg.cfg.InboundTunnelPort,
+		"return")
+
+	// Create a new chain for redirecting outbound traffic to the common Envoy port.
+	// In both chains, 'counter RETURN' bypasses Envoy and 'counter jump IstioRedirectChain'
+	// redirects to Envoy.
+	cfg.ruleBuilder.AppendRule(
+		constants.IstioRedirectChain, constants.IstioProxyNatTable,
```

Comment:
nit: should this be named IstioOutRedirectChain to match IstioInRedirectChain below?